Startup issue on recent macOS on M3

(Note, I don't have a mac myself, I still need to whip up a mini on AWS to reproduce, this is what colleagues have witnessed). 

On macOS on a recent Macbook (thus on Apple's ARM chips), it seems like some of the ODPI structs have different sizes. The initialization check finds a discrepancy between the expected size for the OraData (48 bytes) and the actual size (64 bytes). 

Given that OraDataBuffer does seem to pass the check, the only reasonable hypothesis is some extremely aggressive alignment on mac. 

In any case, I mostly am putting this out there in case anyone else sees this and to figure out whether we're the first to use Oracle.jl on mac/ARM? It sounds unlikely...

(more to follow, but probably not until next year)

```
(.venv) ➜  src git:(eng-2485) ✗ julia oracle_example.jl
ERROR: LoadError: InitError: AssertionError: OraData should have sizeof 48 bytes. Found 64 bytes.
Stacktrace:
  [1] __init__()
    @ Oracle ~/.julia/packages/Oracle/4DDzj/src/Oracle.jl:45
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -27,6 +27,7 @@ jobs:
           - macOS-latest
         arch:
           - x64
+          - aarch64
     steps:
       - uses: actions/checkout@v3
       - uses: julia-actions/setup-julia@v1

--- a/src/types.jl
+++ b/src/types.jl
@@ -1,6 +1,12 @@
 
 # 40 bytes -> sizeof dpiDataBuffer on 64bit arch
-primitive type OraDataBuffer 40 * 8 end
+struct OraDataBuffer
+  a::Int64
+  b::Int64
+  c::Int64
+  d::Int64
+  e::Int64
+end
 
 struct OraData
     is_null::Int32 # Specifies if the value refers to a null value (1) or not (0).

